lib/resque/scheduler.rb

```diff
@@ -339,15 +339,16 @@ def poll_sleep_loop
             sleep(remaining_sleep)
             handle_signals
           rescue Interrupt
-            if @shutdown
-              Resque.clean_schedules
-              release_master_lock
-            end
+            before_shutdown if @shutdown
             break
           end
         end
       end
 
+      def before_shutdown
```

Comment:
Yeah. The master lock is set with a (configurable) timeout of 3 minutes. So, if the master dies for any reason, another available instance will take over as master once the master lock expires in redis.

**Details of why timeout is 3 minutes if you're interested/curious**: the timeout is set to a relatively long duration because it is also a guard against double-scheduling jobs due to clock drift between servers. Consider the case where two instances have their clocks off by 2 minutes and your timeout is say a minute. Then if master dies and the other takes over, you risk double-scheduling jobs